NPE with bazel mod show_repo '<root>'

### Description of the bug:


In any Bazel workspace (even an empty one), run:

```
$ USE_BAZEL_VERSION=last_green bazel mod show_repo '<root>'
 checking cached actions
    Fetching module extension xcode_configure_extension in @@bazel_tools//tools/osx:xcode_configure.bzl; starting
FATAL: bazel crashed due to an internal error. Printing stack trace:
java.lang.RuntimeException: Unrecoverable error while evaluating node 'BZLMOD_REPO_RULE:@@' (requested by nodes )
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:557)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:426)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Unknown Source)
	at java.base/java.util.Optional.of(Unknown Source)
	at com.google.devtools.build.lib.skyframe.BzlmodRepoRuleFunction.checkRepoFromBazelModules(BzlmodRepoRuleFunction.java:159)
	at com.google.devtools.build.lib.skyframe.BzlmodRepoRuleFunction.compute(BzlmodRepoRuleFunction.java:113)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:468)
	... 7 more
```

### Which category does this issue belong to?

External Dependency

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.


_No response_

### Which operating system are you running Bazel on?


_No response_

### What is the output of `bazel info release`?


7.1.0 (or last_green)

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.


_No response_

### What's the output of `git remote get-url origin; git rev-parse HEAD` ?


_No response_

### Is this a regression? If yes, please try to identify the Bazel commit where the bug was introduced.


_No response_

### Have you found anything relevant by searching the web?

_No response_

### Any other information, logs, or outputs that you want to share?

_No response_

--- a/src/main/java/com/google/devtools/build/lib/skyframe/BzlmodRepoRuleFunction.java
+++ b/src/main/java/com/google/devtools/build/lib/skyframe/BzlmodRepoRuleFunction.java
@@ -158,7 +158,7 @@ public final class BzlmodRepoRuleFunction implements SkyFunction {
     if (moduleKey == null) {
       return Optional.empty();
     }
-    return Optional.of(bazelDepGraphValue.getDepGraph().get(moduleKey).getRepoSpec());
+    return Optional.ofNullable(bazelDepGraphValue.getDepGraph().get(moduleKey).getRepoSpec());
   }
 
   @Nullable

--- a/src/test/py/bazel/bzlmod/mod_command_test.py
+++ b/src/test/py/bazel/bzlmod/mod_command_test.py
@@ -507,14 +507,15 @@ class ModCommandTest(test_base.TestBase):
         stderr,
     )
 
+  # fix for https://github.com/bazelbuild/bazel/issues/22587
   def testShowRepoThrowsNonexistentRepo(self):
     _, _, stderr = self.RunBazel(
-        ['mod', 'show_repo', '@@lol'],
+        ['mod', 'show_repo', '<root>'],
         allow_failure=True,
         rstrip=True,
     )
     self.assertIn(
-        "ERROR: In repo argument @@lol: no such repo. Type 'bazel help mod' "
+        "ERROR: In repo argument <root>: no such repo. Type 'bazel help mod' "
         'for syntax and help.',
         stderr,
     )

